src/Components/SysTable/SysTable.js

```diff
@@ -1,183 +1,173 @@
-                        <Tooltip content={<DateFormat date={new Date(data.lastMatchDate)} type='exact' />}>
-                            <span><DateFormat date={new Date(data.lastMatchDate)} type='onlyDate' /></span>
-                        </Tooltip>
-                        : <Tooltip content={intl.formatMessage(messages.noHostHas)}>
-                            <span>{intl.formatMessage(messages.never)}</span>
-                        </Tooltip>
-                },
-                { title: <Link to={`/systems/${data.displayName}`}>{data.totalMatches?.toLocaleString()}</Link> },
-                { title: <Tooltip content={<DateFormat date={new Date(data.lastScanDate)} type='exact' />}>
-                    <span><DateFormat date={new Date(data.lastScanDate)} /></span>
-                </Tooltip> }
-
-            ]
-        }));
-
-        stateSet({ type: 'setRows', payload: rowBuilder(sysTableData) });
-    }, [intl, sysTableData]);
+    const fetchSystems = useFetchSystems({
+        query: GET_SYSTEM_TABLE
+    });
+
+    const mergedColumns = (defaultColumns) =>
+        columns.map((column) => {
+            const isStringCol = typeof column === 'string';
+            const key = isStringCol ? column : column.key;
+            const defaultColumn = defaultColumns.find(
+                (defaultCol) => defaultCol.key === key
+            );
+            return {
+                ...defaultColumn,
+                ...(isStringCol ? { key: column } : column),
+                props: {
+                    ...defaultColumn?.props,
+                    ...column?.props
+                }
+            };
+        });
+
+    const appendDirection = (attributes, direction) =>
+        attributes.map((attribute) => `${attribute}_${direction}`);
+
+    const findColumnByKey = (key) =>
+        (columns || []).find((column) => column.key === key);
 
     return <React.Fragment>
-        <PrimaryToolbar
-            pagination={{
-                itemCount: sysTableData?.hosts?.totalCount || 0,
-                page,
-                perPage: tableVars.limit,
-                onSetPage(e, page) { onSetPage(e, page); },
-                onPerPageSelect(e, perPage) { onPerPageSelect(e, perPage); },
-                isCompact: true
+        <InventoryTable
+            isFullView
+            autoRefresh
+            initialLoading
+            hideFilters={{ all: true, name: false, tags: true }}
+            columns={mergedColumns}
+            // TODO: enable when tag filtering is ready on BE
+            // showTags
+            ref={inventory}
+            fallback={<Spinner />}
+            onLoad={defaultOnLoad(columns, getRegistry)}
+            getEntities={async (
+                _ids,
+                { page = 1, per_page: perPage, orderBy, orderDirection, filters }
+            ) => {
+                const sortableColumn = findColumnByKey(orderBy);
+                const sortBy =
+                sortableColumn && sortableColumn.sortBy
```

Comment:
Can be simplified with safe accessor
```suggestion
                sortableColumn?.sortBy
```